src/libraries/Common/src/Interop/Unix/System.Native/Interop.SocketAddress.cs

```diff
@@ -10,30 +10,39 @@ internal static partial class Interop
     internal static partial class Sys
     {
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetIPSocketAddressSizes")]
+        [SuppressGCTransition]
         internal static extern unsafe Error GetIPSocketAddressSizes(int* ipv4SocketAddressSize, int* ipv6SocketAddressSize);
 
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetAddressFamily")]
+        [SuppressGCTransition]
         internal static extern unsafe Error GetAddressFamily(byte* socketAddress, int socketAddressLen, int* addressFamily);
 
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_SetAddressFamily")]
+        [SuppressGCTransition]
         internal static extern unsafe Error SetAddressFamily(byte* socketAddress, int socketAddressLen, int addressFamily);
 
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetPort")]
+        [SuppressGCTransition]
         internal static extern unsafe Error GetPort(byte* socketAddress, int socketAddressLen, ushort* port);
 
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_SetPort")]
+        [SuppressGCTransition]
         internal static extern unsafe Error SetPort(byte* socketAddress, int socketAddressLen, ushort port);
 
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetIPv4Address")]
+        [SuppressGCTransition]
         internal static extern unsafe Error GetIPv4Address(byte* socketAddress, int socketAddressLen, uint* address);
 
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_SetIPv4Address")]
+        [SuppressGCTransition]
         internal static extern unsafe Error SetIPv4Address(byte* socketAddress, int socketAddressLen, uint address);
 
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetIPv6Address")]
+        [SuppressGCTransition]
```

Comment:
The `SystemNative_GetIPv6Address` might be border line for `SuppressGCTransition`. I don't know the implementation of `memcpy_s` on all platforms and some might do some checking that could take longer than we want. The fact that it is only 16-bytes seems reasonable though. Just something to consider.

Reply:
Same with `SystemNative_SetIPv6Address` below.